src/software/ai/hl/stp/play/shoot_or_pass_play.cpp

```diff
@@ -86,12 +97,41 @@ void ShootOrPassPlay::getNextTactics(TacticCoroutine::push_type &yield,
         auto receiver = std::make_shared<ReceiverTactic>(
             world.field(), world.friendlyTeam(), world.enemyTeam(), pass, world.ball(),
             false);
+
+        auto pass_eval = pass_generator.generatePassEvaluation(world);
+
+        Zones cherry_pick_region_1 = {pass_eval.getBestZonesToCherryPick(
+            world.field(), best_pass_and_score_so_far.pass.receiverPoint())[0]};
+        Zones cherry_pick_region_2 = {pass_eval.getBestZonesToCherryPick(
+            world.field(), best_pass_and_score_so_far.pass.receiverPoint())[1]};
```

Comment:
If you wanted to be really safe, you could have it return a fixed size array and use `set::get<0>(array)` to get a compile-time assertion that the value will exist.

Reply:
unfortunately because std::array needs template arguments and the size of the enum is not known at compile time (we don't use a `COUNT` entry in our `make_enum` implementation), I can't change it to std::array here :(

But i've changed the function name to imply that its just ranking the zones for cherry picking, and updated the javadoc